### DISCUSSION TITLE
Exact settings to get fauxmoEXP to work

### DISCUSSION THREAD

**Discussion by Skybeat** [vintlabs/fauxmoESP#138] [General]
Hello,

it would be very very nice to get the exact settings you use to get fauxmoESP to work wiht echo's.
What i mean is:
- Arduino IDE Versio you use
- ESP Boards version you use
- ESPAsyncTCP version you use
- the exact board settings you use in Arduino IDE
- which ESP you use.

I strugle since the beginning to work with fauxmoESP and can't get it to work. Whenever I look for help, there are other statements which version works with which board version. That is very confusing.

Thank you

**Comment by pvint** [comment]
Here's what I have been using:

## For ESP32:
### Versions:
- FauxmoESP 3.1.2
- Arduino 1.8.13 (I doubt that you'd have any issue with any version of Arduino from the past year or so though)
- AsyncTCP: I use from git https://github.com/me-no-dev/AsyncTCP   There hasn't been much for changes to this in a while, so exact version should not be important

### Boards:
I mostly use my own boards - primarily this one: https://www.vintlabs.com/product/vintlabs-esp32-wroom32-iot-8x2a-pwm-driver-development-kit-4mb-flash-wifi-bluetooth/

### Arduino board settings:
![ArduinoSettingsESP32](https://user-images.githubusercontent.com/608740/102143015-4c56ab00-3e31-11eb-8d1e-84e8fa4d3c72.png)


## For ESP8266:
### Versions:
- FauxmoESP 3.1.2
- Arduino 1.8.13 (I doubt that you'd have any issue with any version of Arduino from the past year or so though)
- ESPAsyncTCP from git https://github.com/me-no-dev/ESPAsyncTCP

### Board
NodeMCU V3

### Arduino Board Settings
![fauxmoESP-minimal | Arduino 1 8 13_109](https://user-images.githubusercontent.com/608740/102143703-54fbb100-3e32-11eb-8cdf-c9d8f1c87e11.png)

## NOTE: 
Recently a few others have had issues with setting `lwIP Variant` to `v2`, so for now I would recommend `v1.4 Higher Bandwidth`

**Reply by mcspr** [reply]
Re:
> so for now I would recommend v1.4 Higher Bandwidth

ESP8266 Core 3.0.0+ removes such option. I can't really tell what the exact issue might be, at a glance, but I'd suggest to take a look at onTCP... functions taking RAW char pointer data as `String`. It is not really valid, as network buffers are not C-strings and are *not* '\0'-terminated.

**Reply by pvint** [reply]
> ESP8266 Core 3.0.0+ removes such option. I can't really tell what the exact issue might be, at a glance, but I'd suggest to take a look at onTCP... functions taking RAW char pointer data as `String`. It is not really valid, as network buffers are not C-strings and are _not_ '\0'-terminated.

I'll have a look at that, thanks.

Be easier to troubleshoot if I could actually reproduce the issue...hopefully I can find a way to do that.

**Reply by HarrisonMylesG** [reply]
Any workarounds? I just put on the latest version of the core and found this out the hard way... nothing working!
